### ISSUE TITLE
Allow passing arguments to AttachCurrentThread

### ISSUE THREAD

**Issue by half-cambodian-hacker-man** [jni-rs/jni-rs#268]
At the moment, we're always sending a null pointer for what would be [`AttachCurrentThread`'s `thr_args`](https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/invocation.html#AttachCurrentThread) - the `JavaVM` struct's `attach_current_thread` doesn't allow passing in any arguments at all.

Currently, I'm using a workaround in which I just call `Thread.currentThread().setName(str)` through some `call_method` gymnastics. It would still be nice to have this functionality in the bindings, since this is something that's super trivial in C, but not possible to do in Rust without dropping into unsafe.

**Comment by dmitry-timofeev** [comment]
Makes sense, understanding which Java threads correspond to which native threads could be useful. [Java 15 spec](https://docs.oracle.com/en/java/javase/15/docs/specs/jni/invocation.html#attachcurrentthread) has more details on the structure (but it also must be in the `jni.h`): 

```c
typedef struct JavaVMAttachArgs {
    jint version;
    char *name;    /* the name of the thread as a modified UTF-8 string, or NULL */
    jobject group; /* global ref of a ThreadGroup object, or NULL */
} JavaVMAttachArgs
```

@stanislav-tkach , what is the best approach to deal with an optional method argument in Rust? Adding a method with a different name, or requiring all usages to pass some "empty" value? In Java we'd just add an overloaded method :)

**Comment by dmitry-timofeev** [comment]
I remember when we implemented the `Executor`, we considered specifying the thread names. If we add this functionality, I'd also suggest to use it in the `Executor`

**Comment by stanislav-tkach** [comment]
I think that a separate method is better than passing an empty value. In situation with multiple optional parameters the best we can do is to create a builder. Unfortunately Rust lacks both functions overload and default parameter values.
